File: AkitaModel/SmartScan/Events/SlotInvalidatedEvent.cs
namespace Skyware.Lis.AkitaModel.SmartScan.Events;

/// <summary>
/// Represents invalidation of a slot.
/// This is an event payload object (SignalR).
/// </summary>
public class SlotInvalidatedEvent : ITypedEvent
{

    /// <summary>
    /// Id of the invalidated slot
    /// </summary>
    public string Id { get; set; }

    /// <inheritdoc/>
    public string MethodName => "slotinvalidated";

}

File: AkitaMotelTS/src/SmartScan/Events/SlotInvalidatedEvent.ts
/**
 * Skyware.Lis.AkitaModel.SmartScan.Events
 */

import { ITypedEvent } from './ITypedEvent';

/**
 * Represents invalidation of a slot.
 * This is an event payload object (SignalR).
 */
export class SlotInvalidatedEvent implements ITypedEvent {
  /**
   * Id of the invalidated slot
   */
  id?: string;

  /** @inheritdoc */
  get methodName(): string {
    return 'slotinvalidated';
  }
}
